llvm/lib/Target/AMDGPU/SIWholeQuadMode.cpp

```diff
@@ -1717,7 +1719,7 @@ bool SIWholeQuadMode::runOnMachineFunction(MachineFunction &MF) {
   LIS->removeAllRegUnitsForPhysReg(AMDGPU::SCC);
 
   // If we performed any kills then recompute EXEC
-  if (!KillInstrs.empty())
+  if (!KillInstrs.empty() || !InitExecInstrs.empty())
     LIS->removeAllRegUnitsForPhysReg(AMDGPU::EXEC);
```

Comment:
This shouldn't be necessary in the first place. Reserved registers, like exec, do not have liveness tracked

Reply:
Well I see stuff like this in the dumps:
```
# After SI Whole Quad Mode
********** INTERVALS **********
EXEC_LO_LO16 [16r,16d:0) 0@16r
EXEC_LO_HI16 [16r,16d:0) 0@16r
EXEC_HI_LO16 [16r,16d:0) 0@16r
EXEC_HI_HI16 [16r,16d:0) 0@16r
```

Reply:
Those shouldn't have been created in the first place. Where did they come from?

Reply:
In this test case they are formed in Register Coalescer when it attempts to merge a copy of EXEC with a virtual and asks for an interval on a EXEC reg unit.

Registered registers still have intervals, they are a sequence of a dead defs, one for each assignment to the register.
You can see this in `LiveIntervals::computeRegUnitRange`.
I do not know the use for this, but assume it is for interference calculations.